File: lib/android/com/hud/RCProgressHUD.java
package com.hud;

import android.app.Activity;
import android.content.Context;
import android.os.Handler;
import android.os.Looper;
import com.avos.avoscloud.AVOSCloud;
import com.bigkoo.svprogresshud.SVProgressHUD;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;
import com.medicalreact.MainActivity;

/**
 * Created by zhuchao on 15/9/29.
 */
public class RCProgressHUD extends ReactContextBaseJavaModule {
    public Activity activity;

    public RCProgressHUD(ReactApplicationContext reactContext,Activity activity) {
        super(reactContext);
        this.activity = activity;
    }
    @Override
    public String getName() {
        return "RCProgressHUD";
    }
    @ReactMethod
    public void dismiss(){

        Runnable myRunnable = new Runnable() {
            @Override
            public void run() {
                SVProgressHUD.dismiss(activity);
            }
        };
        this.runInMainThread(myRunnable);
    }
    @ReactMethod
    public void showWithMaskType(final String maskType){
        final SVProgressHUD.SVProgressHUDMaskType type = this.convMaskType(maskType);
        Runnable myRunnable = new Runnable() {
            @Override
            public void run() {
                SVProgressHUD.showWithMaskType(activity,type);
            }
        };
        this.runInMainThread(myRunnable);
    }
    @ReactMethod
    public void showSuccessWithStatus(final String status){
        Runnable myRunnable = new Runnable() {
            @Override
            public void run() {
                SVProgressHUD.showSuccessWithStatus(activity, status);
            }
        };
        this.runInMainThread(myRunnable);
    }
    @ReactMethod
    public void showSuccessWithStatusAndMaskType(final String status,String maskType){
        final SVProgressHUD.SVProgressHUDMaskType type = this.convMaskType(maskType);
        Runnable myRunnable = new Runnable() {
            @Override
            public void run() {
                SVProgressHUD.showSuccessWithStatus(activity, status,type);
            }
        };
        this.runInMainThread(myRunnable);
    }
    @ReactMethod
    public void showErrorWithStatus(final String status){
        Runnable myRunnable = new Runnable() {
            @Override
            public void run() {
                SVProgressHUD.showErrorWithStatus(activity, status);
            }
        };
        this.runInMainThread(myRunnable);
    }
    @ReactMethod
    public void showErrorWithStatusAndMaskType(final String status,String maskType){

        final SVProgressHUD.SVProgressHUDMaskType type = this.convMaskType(maskType);
        Runnable myRunnable = new Runnable() {
            @Override
            public void run() {
                SVProgressHUD.showErrorWithStatus(activity, status, type);
            }
        };
        this.runInMainThread(myRunnable);
    }
    @ReactMethod
    public void showInfoWithStatus(final String status){
        Runnable myRunnable = new Runnable() {
            @Override
            public void run() {
                SVProgressHUD.showInfoWithStatus(activity, status);
            }
        };
        this.runInMainThread(myRunnable);
    }
    @ReactMethod
    public void showInfoWithStatusAndMaskType(final String status,String maskType){
        final SVProgressHUD.SVProgressHUDMaskType type = this.convMaskType(maskType);
        Runnable myRunnable = new Runnable() {
            @Override
            public void run() {
                SVProgressHUD.showInfoWithStatus(activity, status, type);
            }
        };
        this.runInMainThread(myRunnable);
    }

    public SVProgressHUD.SVProgressHUDMaskType convMaskType(String maskType){
        SVProgressHUD.SVProgressHUDMaskType type = SVProgressHUD.SVProgressHUDMaskType.None;
        if(maskType.equals("Black")){
            type =  SVProgressHUD.SVProgressHUDMaskType.Black;
        }else if(maskType.equals("Clear")){
            type =  SVProgressHUD.SVProgressHUDMaskType.Clear;
        }else if(maskType.equals("Gradient")){
            type =  SVProgressHUD.SVProgressHUDMaskType.Gradient;
        }
        return type;
    }

     public void runInMainThread(Runnable myRunnable){
        Handler mainHandler = new Handler(Looper.getMainLooper());
        mainHandler.post(myRunnable);
    }
}

File: lib/ios/RCProgressHUD.m
//
//  RNProgressHUD.m
//  medicalReact
//
//  Created by zhuchao on 15/9/29.
//  Copyright © 2015年 Facebook. All rights reserved.
//

#import "RCProgressHUD.h"

@implementation RCProgressHUD
RCT_EXPORT_MODULE();
RCT_EXPORT_METHOD(showWithMaskType:(NSString *)maskType){
    dispatch_async(dispatch_get_main_queue(), ^{
        [SVProgressHUD showWithMaskType:[self convMaskType:maskType]];
    });
}
RCT_EXPORT_METHOD(showSuccessWithStatus:(NSString *)status){
    dispatch_async(dispatch_get_main_queue(), ^{
        [SVProgressHUD showSuccessWithStatus:status];
    });
}

RCT_EXPORT_METHOD(showSuccessWithStatusAndMaskType:(NSString *)status type:(NSString *)maskType){
    dispatch_async(dispatch_get_main_queue(), ^{
        [SVProgressHUD showSuccessWithStatus:status maskType:[self convMaskType:maskType]];
    });
}

RCT_EXPORT_METHOD(showErrorWithStatus:(NSString *)status){
    dispatch_async(dispatch_get_main_queue(), ^{
        [SVProgressHUD showErrorWithStatus:status];
    });
}

RCT_EXPORT_METHOD(showErrorWithStatusAndMaskType:(NSString *)status type:(NSString *)maskType){
    dispatch_async(dispatch_get_main_queue(), ^{
        [SVProgressHUD showErrorWithStatus:status maskType:[self convMaskType:maskType]];
    });
}

RCT_EXPORT_METHOD(showInfoWithStatus:(NSString *)status){
    dispatch_async(dispatch_get_main_queue(), ^{
        [SVProgressHUD showInfoWithStatus:status];
    });
}

RCT_EXPORT_METHOD(showInfoWithStatusAndMaskType:(NSString *)status type:(NSString *)maskType){
    dispatch_async(dispatch_get_main_queue(), ^{
        [SVProgressHUD showInfoWithStatus:status maskType:[self convMaskType:maskType]];
    });
}

RCT_EXPORT_METHOD(dismiss){
    dispatch_async(dispatch_get_main_queue(), ^{
        [SVProgressHUD dismiss];
    });
}

-(SVProgressHUDMaskType)convMaskType:(NSString *)maskType{
    SVProgressHUDMaskType type = SVProgressHUDMaskTypeNone;
    if ([maskType isEqualToString:@"Black"]){
        type = SVProgressHUDMaskTypeBlack;
    }else if ([maskType isEqualToString:@"Clear"]){
        type = SVProgressHUDMaskTypeClear;
    }else if ([maskType isEqualToString:@"Gradient"]){
        type = SVProgressHUDMaskTypeGradient;
    }
    return type;
}
@end
